src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralCommandHandler.cs

```diff
@@ -92,61 +92,50 @@ public bool ExecuteCommandWorker(ReturnKeyCommandArgs args)
                 }
             }
 
-            IndentationOptions? lazyOptions = null;
+            var document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null)
+            {
+                return false;
+            }
+
+            var parsedDocument = ParsedDocument.CreateSynchronously(document, CancellationToken.None);
+            var options = subjectBuffer.GetIndentationOptions(_editorOptionsService, parsedDocument.LanguageServices, explicitFormat: false);
 
             // We now go through the verified string literals and split each of them.
             // The list of spans is traversed in reverse order so we do not have to
             // deal with updating later caret positions to account for the added space
             // from splitting at earlier caret positions.
             foreach (var span in spans.Reverse())
             {
-                if (!SplitString(textView, subjectBuffer, span.Start.Position, ref lazyOptions, CancellationToken.None))
+                using var transaction = CaretPreservingEditTransaction.TryCreate(
+                CSharpEditorResources.Split_string, textView, _undoHistoryRegistry, _editorOperationsFactoryService);
+
+                var splitter = StringSplitter.TryCreate(parsedDocument, span.Start.Position, options, CancellationToken.None);
+                if (splitter?.TrySplit(out var newRoot, out var newPosition) != true)
                 {
                     return false;
                 }
-            }
-
-            return true;
-        }
```

Comment:
is it possible to keep this split (no pun intended).  it was kinda nice to have the loop out here, and the inner helper which just handles one split.

Reply:
It's actually uglier that way now since each iteration updates the document, so I'd need to pass it by-ref.